python/lsst/summit/extras/plotting/focusSweep.py

```diff
@@ -0,0 +1,282 @@
+# This file is part of summit_extras.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = [
+    "collectSweepData",
+    "fitSweepParabola",
+    "plotSweepParabola",
+]
+
+
+import numpy as np
+from matplotlib.figure import Figure
+
+from lsst.summit.utils.efdUtils import efdTimestampToAstropy, getMostRecentRowWithDataBefore
+
+PLATESCALE = 0.2  # arcsec / pixel
+
+
+def collectSweepData(records, consDbClient, efdClient):
+    """
+    Populate focus sweep table.
+
+    Parameters
+    ----------
+    records : list of `DimensionRecord`
+        Records spanning focus sweep.
+    consDbClient: `ConsDbClient`
+        Consolidated database client
+    efdClient: `EfdClient`
+        Engineering facilities database client
+
+    Returns
+    -------
+    table : astropy.table.Table
+        Table containing hexapod sweep motions and quick look PSF measurements.
+    """
+    visitString = ",".join(str(r.id) for r in records)
+    data = consDbClient.query(
+        "SELECT "
+        "visit_id as visit_id, "
+        "n_inputs as n_inputs, "
+        "psf_sigma_median as sigma, "
+        "psf_ixx_median as ixx, "
+        "psf_iyy_median as iyy, "
+        "psf_ixy_median as ixy "
+        f"from cdb_lsstcomcamsim.visit1_quicklook WHERE visit_id in ({visitString}) "
+        "ORDER BY visit_id"
+    )
+    data["seqNum"] = data["visit_id"] % 10000
```

Comment:
It looks like you're already expecting to be given `DimensionRecord`s for all of these visits, and it'd be a lot less fragile to pull these out in the loop below as

```
row["seqNum"] = record.seq_num
```

Reply:
Agreed. We don't generally approve of people parsing the visit_id because the form depends directly on the visit system and that isn't being checked here. It works because consDB is using a single visit system.

Reply:
I totally agree (and this is Josh's code btw). I think the issue here is that the data can come out of consDB in any order (at least in theory), so either we need to build the table up by looping over the `DimensionRecords`, which is inefficient, but would mean we have the seqNum from the loop, or we need to get it from something in the table already, otherwise we don't actually know which DimensionRecord it came from. Really, that should just already be a column in the database though, and then this would be a non-issue (here and many other places), but we don't have the wide views yet and KT doesn't want to replicate things anywhere.

Reply:
Oh, I see what Jim means about the "loop below" - I missed that. OK, that's gross by makes sense. OK, I'll do that 👍